src/service/sync/ldap-sync-service.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2024  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *  @license
+ */
+
+import { SyncService } from './sync-service';
+import User, { TermsOfServiceStatus, UserType } from '../../entity/user/user';
+import { Client } from 'ldapts';
+import ADService from '../ad-service';
+import LDAPAuthenticator from '../../entity/authenticator/ldap-authenticator';
+import RoleManager from '../../rbac/role-manager';
+import { EntityManager } from 'typeorm';
+
+export default class LdapSyncService extends SyncService {
+
+  // We only sync organs and members.
+  targets = [UserType.ORGAN, UserType.MEMBER];
+
+  private readonly ldapClient: Client;
+
+  private readonly roleManager: RoleManager;
+
+  constructor(manager: EntityManager, ldapClient: Client, roleManager: RoleManager) {
```

Comment:
Manager is optional in the base class `SyncService`, but is not optional in this derivative class.